inference/sentence_transformers_model.py

```diff
@@ -55,12 +64,87 @@ def __init__(
         self.model = SentenceTransformer(**sentence_transformers_kwargs)
         self.model = self.model.eval()
 
+        self.batching_enabled = True  # Always true since encode() handles batching
+        self.max_batch_size = 32  # Default from encode() method
+        self.batch_wait_timeout_s = 0.1
+
     @web_app.get("/model_device")
     def model_device(self) -> str:
         return str(self.model.device)
 
+    @serve.batch(max_batch_size=32, batch_wait_timeout_s=0.1)
+    async def _batch_infer(
+        self, requests: List[BatchableInferenceRequest]
+    ) -> List[dict]:
+        try:
+            logger.info(f"Starting batch call with {len(requests)} requests")
+
+            # Group by kwargs since encode() supports different options
+            current_group: List[Any] = []
+            current_kwargs: str | None = None
+            results: List[dict] = []
+
+            for request in requests:
+                # Handle None kwargs by converting to empty dict
+                kwargs_to_serialize = (
+                    request.kwargs if request.kwargs is not None else {}
+                )
+                kwargs_key = json.dumps(kwargs_to_serialize, sort_keys=True)
+
+                if kwargs_key != current_kwargs and current_group:
+
+                    if current_kwargs is None:
+                        raise ValueError(
+                            "current_kwargs should not be None at this point"
+                        )
```

Comment:
If this is a "sanity check" for your own benefit, it should be an `assert`, e.g.

```suggestion
                    assert current_kwargs is not None
```